File: snippets/fsharp/VS_Snippets_CLR_System/system.BitConverter.Class/FS/example1.fs
module example1

// <Snippet3>
open System

let value = -16
let bytes = BitConverter.GetBytes value

// Convert bytes back to int.
let intValue = BitConverter.ToInt32(bytes, 0)
printfn $"""{value} = {intValue}: {if value.Equals intValue then "Round-trips" else "Does not round-trip"}"""

// Convert bytes to UInt32.
let uintValue = BitConverter.ToUInt32(bytes, 0)
printfn $"""{value} = {uintValue}: {if value.Equals uintValue then "Round-trips" else "Does not round-trip"}"""


// The example displays the following output:
//       -16 = -16: Round-trips
//       -16 = 4294967280: Does not round-trip
// </Snippet3>

File: snippets/csharp/System/BitConverter/Overview/example1.cs
// <Snippet3>
using System;

public class BitConverterExample
{
    public static void Main()
    {
        int value = -16;
        byte[] bytes = BitConverter.GetBytes(value);

        // Convert bytes back to int.
        int intValue = BitConverter.ToInt32(bytes, 0);
        Console.WriteLine($"{value} = {intValue}: {(value.Equals(intValue) ? "Round-trips" : "Does not round-trip")}");
        // Convert bytes to UInt32.
        uint uintValue = BitConverter.ToUInt32(bytes, 0);
        Console.WriteLine($"{value} = {uintValue}: {(value.Equals(uintValue) ? "Round-trips" : "Does not round-trip")}");
    }
}
// The example displays the following output:
//       -16 = -16: Round-trips
//       -16 = 4294967280: Does not round-trip
// </Snippet3>
